vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java

```diff
@@ -514,6 +542,93 @@ else if (target.fastMatchLocation && target.targetType == GlobalCommandTarget.Ta
           }
         }
       }
+
+      // If a specific X, Y target has been specified AND a valid target map, then we can go direct to the Qtree index to find those pieces
+      else if (target.fastMatchLocation && target.targetType == GlobalCommandTarget.Target.XY && targetFastMap != null && usePieceIndexing) {
+
+        int x = 0;
+        int y = 0;
+
+        try {
+          x = Integer.parseInt(fastX);
+        }
+        catch (Exception ignored) {
+          x = 0;
+        }
+
+        try {
+          y = Integer.parseInt(fastY);
+        }
+        catch (Exception ignored) {
```

Comment:
Should catch `NumberFormatException` here. Anything broader and it's swallowing a bug.